src/main/java/org/hiero/sketch/table/LazyMembership.java

```diff
@@ -41,7 +40,7 @@ public LazyMembership(final IMembershipSet baseMap, final Predicate<Integer> fil
         if (filter == null) throw new NullArgumentException("PartialMembershipDense cannot be " +
                 "instantiated with a null filter");
         this.baseMap = baseMap;
-        this.rowCount = baseMap.getSize(false);
+        this.rowCount = 0;
```

Comment:
Currently, when we iterate over it for getSize() we just remember that we did it and don't do it again. I'm not sure it is worthwhile to instantiate a SparseMembership. The main difference is space. I'm thinking of a scenario where the baseMap is fullMembership and the predicate is fairly sparse. In this case LazyMembership will take O(1) space instead of O(n). There is always the possibility to convert it to a sparseMembership (via the sparseMembership constructor).